packages/api/models/base.js

```diff
@@ -281,6 +281,15 @@ class Manager {
     return this.dynamodbDocClient.batchWrite(params).promise();
   }
 
+  addTimeStampsToItem(item) {
+    const now = Date.now();
+    return {
+      createdAt: now,
+      ...item,
+      updatedAt: now,
+    };
```

Comment:
I think if you make this change you don't need the per model overrides? Then all you have to ensure is that `item.createdAt` is equal to the value from the original record for PUTs, and not an arbitrarily created timestamp. And you don't even need to set `updatedAt` on the incoming `item` from the endpoint code

```suggestion
    const now = Date.now();
    return {
      ...item
      createdAt: item.createdAt || now,,
      updatedAt: now,
    };
```

Reply:
One immediate comment: 

>  And you don't even need to set updatedAt on the incoming item from the endpoint code

That was done to have that date set at the same time, rather than differ between dynamo and RDS.  

That said.... 

I'm not clear on what's functionally different in the above snippet: 

For example: 
```
> object = { foo : 'bar' } 
{ foo: 'bar' }
> updateObject = { foo: 'notbar', ...object }
{ foo: 'bar' }
> updateObject2 = { ...object, foo: object.foo || 'notbar' }
{ foo: 'bar' }
```

The change here and in the overridden methods is making updatedAt a set item by the endpoint logic, and not enforced by the model -- this allows a consistent value to be set explicitly. 

I suppose we could avoid all of this override nonsense if this applied to *all* of the models.

Reply:
> I suppose we could avoid all of this override nonsense if this applied to all of the models.

Yeah, that's the point - I think if we wrote the method this way then we wouldn't need any of the overrides. The refactor behaves the same as previously written except that it defaults to using `item.createdAt` if it's already set, otherwise it defaults to now

Reply:
The intent was to make the minimal change required to fix the bug, and avoid the overhead in changing the logic in models that are accessed outside the endpoints given this code is scheduled for removal.        That was perhaps a mistake - I can do some due diligence in seeing what that would do to, say, the granules write logic.

Reply:
476ccf1